### PR TITLE
Update fenix for new megazord setup

### PR DESCRIPTION
**Pull request by thomcc**
Depends on https://github.com/mozilla-mobile/android-components/pull/3806

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/fenix/blob/master/CHANGELOG.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### CODE REVIEW

**Comment by thomcc** [comment]
CC @hawkinsw After this you can defer JNA initialization (by deferring HTTP client init and logging init) the way you wanted to so long ago.

File: build.gradle [hunk 1]
```diff
@@ -23,6 +23,7 @@ plugins {
 allprojects {
     repositories {
         google()
+        mavenLocal()
```
Comment (grigoryk) [inline build.gradle]: remove this

File: buildSrc/src/main/java/Dependencies.kt [hunk 2]
```diff
@@ -39,7 +39,7 @@ object Versions {
     // The version number below tracks the application-services version
     // that we depend on directly for tests, and it's important that it
     // be kept in sync with the version used by android-components above.
-    const val mozilla_appservices = "0.34.0"
+    const val mozilla_appservices = "0.35.0"
```
Comment (grigoryk) [inline buildSrc/src/main/java/Dependencies.kt]: bump to 0.35.4 (same for a-c pr)

File: build.gradle (hunk 1 above)
Comment (thomcc) [reply build.gradle]: Whoops!

File: buildSrc/src/main/java/Dependencies.kt (hunk 2 above)
Comment (thomcc) [reply buildSrc/src/main/java/Dependencies.kt]: I'll wait for 0.35.5 which we're going to cut in a few minutes.

**Comment by hawkinsw** [comment]
> CC @hawkinsw After this you can defer JNA initialization (by deferring HTTP client init and logging init) the way you wanted to so long ago.

This is great! Thank you so much for the hard work! @csadilek

File: buildSrc/src/main/java/Dependencies.kt (hunk 2 above)
Comment (grigoryk) [reply buildSrc/src/main/java/Dependencies.kt]: And now it's 0.36!

**Comment by grigoryk** [comment]
https://github.com/mozilla-mobile/android-components/pull/3806 landed in a-c 7.0.0. We'll need this PR when we update to 7.0.0.

**Comment by grigoryk** [comment]
I think this needs at least two changes:
- add the lost-in-a-rebase `implementation Deps.mozilla_support_rusthttp`
- change a-c version to `7.0.0-SNAPSHOT`

File: buildSrc/src/main/java/Dependencies.kt [hunk 3]
```diff
@@ -33,13 +33,13 @@ object Versions {
...
 
-    const val mozilla_android_components = "6.0.2"
+    const val mozilla_android_components = "7.0.0-SNAPSHOT"
     // Note that android-components also depends on application-services,
     // and in fact is our main source of appservices-related functionality.
     // The version number below tracks the application-services version
     // that we depend on directly for tests, and it's important that it
     // be kept in sync with the version used by android-components above.
-    const val mozilla_appservices = "0.34.0"
+    const val mozilla_appservices = "0.36.0"
```
Comment (grigoryk) [inline buildSrc/src/main/java/Dependencies.kt]: Can you update the comment just above? we now also depend on it for the megazords.

File: buildSrc/src/main/java/Dependencies.kt [hunk 4]
```diff
@@ -200,7 +201,10 @@ object Deps {
     const val robolectric = "org.robolectric:robolectric:${Versions.robolectric}"
     const val fragment_testing = "androidx.fragment:fragment-testing:${Versions.androidx_testing}"
     const val androidx_junit = "androidx.test.ext:junit:${Versions.androidx_test_ext}"
-    const val places_forUnitTests = "org.mozilla.appservices:places-forUnitTests:${Versions.mozilla_appservices}"
+    // for specifying as a substitution.
+    const val full_megazord = "org.mozilla.appservices:full-megazord:${Versions.mozilla_appservices}"
```
Comment (grigoryk) [inline buildSrc/src/main/java/Dependencies.kt]: you don't need this, right?

File: buildSrc/src/main/java/Dependencies.kt (hunk 3 above)
Comment (thomcc) [reply buildSrc/src/main/java/Dependencies.kt]: It still seems accurate to me, what would you prefer it to say?

File: buildSrc/src/main/java/Dependencies.kt (hunk 4 above)
Comment (thomcc) [reply buildSrc/src/main/java/Dependencies.kt]: Good catch.

**Pull request merged**